Il2CppInterop.Generator/Contexts/AssemblyRewriteContext.cs

```diff
@@ -51,10 +51,12 @@ public TypeRewriteContext GetContextForNewType(TypeDefinition type)
 
     public void RegisterTypeRewrite(TypeRewriteContext context)
     {
-        if (context.OriginalType != null)
+        var exists = context.OriginalType != null && myOldTypeMap.ContainsKey(context.OriginalType);
+        if (context.OriginalType != null && !exists)
             myOldTypeMap[context.OriginalType] = context;
         myNewTypeMap[context.NewType] = context;
-        myNameTypeMap[(context.OriginalType ?? context.NewType).FullName] = context;
+        if (!exists)
+            myNameTypeMap[(context.OriginalType ?? context.NewType).FullName] = context;
     }
```

Comment:
The `exists` check feels really overcomplicated, looking at the usings of it wouldn't it be easier to do something like:
```cs
public void RegisterTypeContext(TypeRewriteContext context)
{
    myNewTypeMap.Add(context.NewType, context);
}

public void RegisterTypeRewrite(TypeRewriteContext context)
{
    if (context.OriginalType != null)
        myOldTypeMap.Add(context.OriginalType, context);
    RegisterTypeContext(context);
    myNameTypeMap.Add((context.OriginalType ?? context.NewType).FullName, context);
}
```
and call `RegisterTypeContext` in `Pass12CreateGenericNonBlittableTypes`?